mod/01_core/set/all/fetch.rb

```diff
@@ -55,7 +55,9 @@ def fetch mark, opts={}
       if opts[:new]
         return card.renew(opts) if !clean_cache_opts? opts
       else
-        return unless !opts[:skip_virtual] && card.virtual?
+        card.include_set_modules unless opts[:skip_modules]  # need to load modules here to call the right virtual? method 
+        opts[:skip_modules] = true                           # don't load twice
+        return if opts[:skip_virtual] || !card.virtual?
```

Comment:
If we're skipping virtual, we should return right away and not run #include_set_modules. 

> BTW, does a code-virtual card have a type?

Every card has a type.  These cards can be of any type.  It just means #virtual? is overridden in a set module.  Am I misunderstanding the question?

> I don't think you need to set :skip_modules. The loading of modules is always conditioned on them being already loaded.

Agree. See `@set_mods_loaded`.